drivers/sensor/hx711/hx711.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2020 Kalyan Sriram <kalyan@coderkalyan.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT avia_hx711
+
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+
+#include "hx711.h"
+
+LOG_MODULE_REGISTER(HX711, CONFIG_SENSOR_LOG_LEVEL);
+
+static void hx711_convert_data(struct sensor_value *val, double raw_val,
+		double offset, double scale)
+{
+	double conv_val = (raw_val * scale) - offset;
```

Comment:
IMHO, the math here should be:

`
double conv_val = (raw_val - offset) / scale;
`

Validated it with four load cells like [these](https://www.sparkfun.com/products/10245) , extracted from digital glass bathroom scale. Setting offset to -87900.0 and scale to -20.0, I've got correct readings around zero grams with empty scale and with various other loads at least up to 70kg.

**Empty scale:**
[0:28:26.980]: calibrated: -0.750000 grams, raw: -87985.000000 grams
[0:28:27.168]: calibrated: 0.850000 grams, raw: -88017.000000 grams
[0:28:27.355]: calibrated: -0.350000 grams, raw: -87993.000000 grams

**With 500g load**
[0:30:08.096]: calibrated: 504.050000 grams, raw: -98081.000000 grams
[0:30:08.283]: calibrated: 501.100000 grams, raw: -98022.000000 grams
[0:30:08.471]: calibrated: 502.600000 grams, raw: -98052.000000 grams